### ISSUE TITLE
LFO Final Report Review Page

### ISSUE THREAD

**Issue by zoeyli-46** [bcgov/cas-reporting#398]
#### Description
Before Industrial Reporters get to the "submit report" page, we need to show them a final review page that displays all the fields they have input into the system. This allows them to print and share the full report before submitting. 

The LFO final review page differs from the SFO page by having the facility grid which will open the facility reports in a separate tab 

#### Acceptance Criteria
Given I am an Industrial Reporter who has completed my annual report
Before I am sent to the submit page, I want to see a full list of all fields for my annual report
So that I can review them before submission. 

As an LFO, Given I am on the final review page, I see the following in read-only mode:
- Operation Information
- **Facilities Information Sections**
- Compliance summary
[LFO Wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=24852-5561&t=Z9nk3eWKE8pnB7dZ-4)

~I see a table of my facilities with an action column
<img width="503" alt="Image" src="https://github.com/user-attachments/assets/49d8d0cb-6691-47bd-952c-eef1df0e9ff0" />
When I click onto View Details for a row,
Then it will open that facility report in a new tab in read only mode
[Wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=24902-6392&t=xEvinZgFOSvauzge-4)~

The user can click the following buttons: Back, Continue, Download (not in MVP)
- [ ] The "Back button" will take the user to the Compliance Summary page
- [ ] The "Continue button" will take the user to the Verification page
- [ ] The "Download..." button will download the page to a PDF (not in MVP)

Hide the following pages/sections in final review:
- [ ] Skip the [Facility success page](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=16744-22148&t=s2yACWJ9iS79p0vg-4)

#### Development Checklist:

- [ ] Create review-submit page similar to `/bciers/apps/reporting/src/app/bceidbusiness/industry_user/reports/[version_id]/review-operator-data`
- [ ] Create RJSF Schema with all steps and foldable groups
- [ ] Create UISchema
- [ ] (Maybe create accordion widget or edit css depending on review)
- [ ] Link redirect on previous compliance summary review page 
- [ ] Perhaps create endpoint/service to grab all report data in one blob for RJSF if needed
- [ ] Add tests for backend endpoint
- [ ] Add tests for frontend page
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Blocked By
#149

**Comment by patriciarussellCAS** [comment]
From refinement: to book a dev/design sync to finalize ticket.

**Comment by fviduya** [comment]
I have tested this in the DEV environment, [here is the test document for reference](https://buttoninc-my.sharepoint.com/:w:/g/personal/francesca_button_is/EYab8AJHLdZDnIwmldnCEKwBFN07H0RbD4mz3KTfywxzcw?e=7lInCh).

I can verify that:
- [x] When submitting a report for an LFO, in the Final Review page, I see the full list of all fields for my annual report to review before submission
- [x] The Final Review page is in read-only mode and consists data from: Operation Information, Facilities Information Sections, Additional Information, and Compliance Summary  

There are a few things I noticed during testing which I'll be creating separate tickets for:
1. In SFO, clicking ```Back``` from the Compliance Summary page redirects to the Operation Emission Summary page
2. SFO's Final Review page is the same as LFO's wherein the report for the facility is hidden in a collapsible content. Previously, SFO's Final Review page showed the whole report by default after the page loads and no click was needed to see report entered for facility - see reference 1 below - It still shows the correct data but would like to get your input on this @patriciarussellCAS / @vesselak 
3. In the Final Review page both for SFO and LFO, all the gas types are listed instead of just the first 6, but in the Non-attributable emissions page only the first 6 are listed - see reference 2 below.
4. ```Save``` button in the activities form does not seem to be saving, it's showing Success message in the button but if you go out and go back in, it will show the old data still.
5. When I misclick a source type for an activity, and tick it off, then continue with reporting the data I want to report, in the Final Review page that misclicked source type is displayed. See page 22 of the linked document. 
6. After finishing the report for a facility, when I click ```Continue``` on an already completed report, the activity forms are blank, but when I continue up until I get to the Final Review page again, the data I previously put is listed there.
7. Upon rechecking the issue above, found also that after completing a report for a facility, when I click ```Continue``` to go back to the facility report, the activities selected defaults back to the Reporting Activities selected in the Operation Information page. 
8. After finishing a report for a facility, and going back in again by clicking the ```Continue``` button, the Emissions Summary page seems to be accumulating data instead of it being exclusive to just the facility you are submitting a report for. 
9. After making corrections in the report, Operation Emission Summary page seems to be accumulating (previous + new calculation)
10. After completing multiple reports, then going back in on a report for a facility, the Allocation of Emissions page now shows that emissions must be allocated for a specific emission category that was applicable to the previous facilities reported for and not for the current one.

**Reference 1: LFO vs SFO report view** 
| SFO | LFO |
| ---- | ---- |
| ![Image](https://github.com/user-attachments/assets/165ea18c-e443-46ec-a981-c5df75675ee3) |  ![Image](https://github.com/user-attachments/assets/7136889d-dde3-4013-ae81-9d44e06fbece) |
 

**Reference 2: Final Review page showing the Gas Types**
![Image](https://github.com/user-attachments/assets/fac4fdac-2ca7-4cdb-b8fa-21c115cc1b35)

For items 6-10, I can create a single ticket to have it tackled and create a document to help with the replication as well since those require quite a bit of steps to actually hit. Then separate tickets for the others to have them isolated. Moving to Pending PO for review of test and notes. Thank you!

**Comment by patriciarussellCAS** [comment]
Thanks so much @fviduya. Here are some of my notes: 
1.	In SFO, clicking Back from the Compliance Summary page redirects to the Operation Emission Summary page
	

> I’ve flagged this issue before, but can’t seem to find the bug ticket. We’ll need to create a new one for this. 

2.	SFO's Final Review page is the same as LFO's wherein the report for the facility is hidden in a collapsible content. Previously, SFO's Final Review page showed the whole report by default after the page loads and no click was needed to see report entered for facility - see reference 1 below - It still shows the correct data but would like to get your input on this [@patriciarussellCAS](https://github.com/patriciarussellCAS) / [@vesselak](https://github.com/vesselak)
	

> This wasn’t the intention, and should be fixed, but is still technically functional so I agree we should make a bug ticket for it, but it should be a low priority ticket for now. 

3.	In the Final Review page both for SFO and LFO, all the gas types are listed instead of just the first 6, but in the Non-attributable emissions page only the first 6 are listed - see reference 2 below.

> I’ll flag this for developers. Dylan hid all the other gas types so that they don’t display on the Non-attrib page, but whatever he did must not have carried over to the final review page. low priority ticket. 

4.	Save button in the activities form does not seem to be saving, it's showing Success message in the button but if you go out and go back in, it will show the old data still.

> I was unable to recreate this. If it's still happening for you, then would need a high-priority bug ticket. 

5.	When I misclick a source type for an activity, and tick it off, then continue with reporting the data I want to report, in the Final Review page that misclicked source type is displayed. See page 22 of the linked document.

> I wasn't able (yet) to click through all the way to see if this happened to me. Let's make this a medium priority ticket.  

6.	After finishing the report for a facility, when I click Continue on an already completed report, the activity forms are blank, but when I continue up until I get to the Final Review page again, the data I previously put is listed there.

> This didn’t happen for me. When I clicked continue on an already complete report, the forms were still filled in as I had made them. I didn’t check all the way to the final review page though. Are you able to re-test to confirm? 

7.	Upon rechecking the issue above, found also that after completing a report for a facility, when I click Continue to go back to the facility report, the activities selected defaults back to the Reporting Activities selected in the Operation Information page.

> This didn’t happen for me. When I clicked on a complete report the activities selected remained as I had changed them. Are you able to re-test? 

8.	After finishing a report for a facility, and going back in again by clicking the Continue button, the Emissions Summary page seems to be accumulating data instead of it being exclusive to just the facility you are submitting a report for.

> This didn't happen for me, when I updated my emissions data on an activity page, the new emissions total carried over. but I only tried it once, and we definitely want to make sure this isn't accumulating data in any way so would definitely be a high priority bug to deal with. 

9.	After making corrections in the report, Operation Emission Summary page seems to be accumulating (previous + new calculation)

> Another one I was unable to confirm in my own test, but will try again tomorrow. If this is the case, we'll need to flag this as a critical, high-priority bug as this will affect their compliance obligation. We may need a senior dev to investigate.  

10.	After completing multiple reports, then going back in on a report for a facility, the Allocation of Emissions page now shows that emissions must be allocated for a specific emission category that was applicable to the previous facilities reported for and not for the current one.

> That's strange. I was unable to confirm this so will have to check another time. BUT I did find a separate bug. When I selected a medium facility, and then did NOT select any products on the Production Data page, I was still unable to get past the Allocation of Emissions page. Medium and Small facilities should be allowed to report NO products, which means they do NOT need to allocate emissions (since there is no product to allocate to). So when on the allocation of emissions page, if no products were selected and the user selects "Not applicable" for methodology, they should be able to proceed to the next page. The current situation is that save & continue are disabled, despite the fact that there's no way to allocate emissions. 

It might make sense to separate out the allocation of emissions bug(s) into it's own ticket, but agree the others could be put together. Thanks!

**Comment by fviduya** [comment]
Thanks, @patriciarussellCAS! I'll be retesting the others once the environment's up again. I've created bcgov/cas-reporting#585 already for first issue flagged here.

**Comment by zoeyli-46** [comment]
Design notes: Could we have a "disabled mode" for checkboxes and radio buttons where they are grayed out? Currently it behaves like its clickable, which may trick users into thinking they can edit things on the final review page.

<img width="823" alt="Image" src="https://github.com/user-attachments/assets/41d07715-8eb4-4803-bebc-160b9ee11fb0" />

The units under compliance summary appear far from the value, is it possible to change the spacing to 10px between the value and the unit? 

<img width="696" alt="Image" src="https://github.com/user-attachments/assets/a8f748d0-c2ea-47e2-b594-c6f6e1fcaca6" />

**Comment by patriciarussellCAS** [comment]
Thanks @zoeyli-46 could we make a design backlog ticket for these items? I would also add to the list this issue. 
Anything they selected in a dropdown menu shows up as very very light grey. It's barely visible! 

![Image](https://github.com/user-attachments/assets/fb8e5ca0-18c5-49c5-b047-de5102e44062)

**Comment by zoeyli-46** [comment]
Created bcgov/cas-registration#5017 for design backlog items

**Comment by patriciarussellCAS** [comment]
@fviduya just wanted to flag that I *think* the allocation of emissions bug I mentioned above will be dealt with through bcgov/cas-reporting#522

**Comment by fviduya** [comment]
Thanks, @patriciarussellCAS! I was trying to recreate the issues regarding the Emission Summary page and Operation Emission Summary page (issues 6-10) but I couldn't quite hit it yet even after following my previous steps. I did however hit the other issues I found which I logged in separate tickets. 

✅  For item 1, I've created bcgov/cas-reporting#585 - issue here has been fixed already by PR from bcgov/cas-reporting#547/#505
For item 2, I've created bcgov/cas-reporting#586 
For item 3, I've created bcgov/cas-reporting#592
For item 4, I was able to recreate this so I've created bcgov/cas-reporting#590. I'm able to replicate this in the DEV env
For item 5, I've created bcgov/cas-reporting#591

**Comment by vesselak** [comment]
Reviewed documentation and accepting ticket with the deficiencies referenced in this ticket and created as part of the testing for bcgov/cas-reporting#398
